frontend/tests/controllers/UserRegistrationTest.php

```diff
@@ -186,4 +186,33 @@ public function testUserParentalTokenNotGeneratedDueInvalidParameters() {
             $this->assertSame('parameterInvalid', $e->getMessage());
         }
     }
+
+    /**
+     * User 13 and its account link to the parent account
+     * on verification of parental token
+     */
+    public function testUse13LinkedToParentAccountWhenTokenVerificationDone() {
+        $defaultDate = strtotime('2022-01-01T00:00:00Z');
+        \OmegaUp\Time::setTimeForTesting($defaultDate);
+        // Create a 10 years-old user
+        ['identity' => $identity] = \OmegaUp\Test\Factories\User::createUser(
+            new \OmegaUp\Test\Factories\UserParams([
+                'birthDate' => strtotime('2012-01-01T00:00:00Z'),
+            ]),
+        );
+        $login = self::login($identity);
+
+        $payload = \OmegaUp\Controllers\User::getVerificationParentalTokenDetailsForTypeScript(
+            new \OmegaUp\Request([
+                'auth_token' => $login->auth_token,
+            ])
+        )['templateProperties']['payload'];
+        //verify token belongs to child accounts ---?
+
+        //assert the message.
+        $this->assertEquals(
+            $payload['hasParentalVerificationToken'],
+            "Parent's account is linked to their child's account"
+        );
```

Comment:
The field `hasParentalVerificationToken` should return a boolean, so you could use the following assertion:
```suggestion
        $this->assertTrue($payload['hasParentalVerificationToken']);
```

Also you could add a new assertion, where both users are linked by `email_id`:
```php

        $updatedUser = \OmegaUp\DAO\Users::getByPK($user->user_id);

        // Assert both accounts are linked by email_id
        $this->assertSame(
            $updatedUser->parent_email_id,
            $parentUser->main_email_id
        );
```

Reply:
Oh yes! Okay done.